tests/python/rest_api/test_tasks.py

```diff
@@ -2819,42 +2820,519 @@ def test_import_annotations(self, task_kind, annotation_kind, expect_success):
+                            "frame": 3,
+                            "points": [1.0, 2.0, 3.0, 2.0],
+                            "keyframe": True,
+                        },
+                    ],
+                    "elements": [],
+                }
+            ],
+        }
+
+        original_annotations, imported_annotations = self.delete_annotation_and_import_annotations(
+            task_id, annotations, format_name, dataset_file
+        )
+
+        self.compare_original_and_import_annotations(original_annotations, imported_annotations)
+
+    @pytest.mark.parametrize("format_name", ["Datumaro 1.0", "COCO 1.0", "PASCAL VOC 1.1"])
+    def test_export_and_import_tracked_format_with_outside_without_keyframe(self, format_name):
+        task_id = 14
+        dataset_file = self.tmp_dir / (format_name + "outside_without_keyframe_source_data.zip")
+        annotations = {
+            "shapes": [],
+            "tracks": [
+                {
+                    "frame": 0,
+                    "shapes": [
+                        {
+                            "type": "rectangle",
+                            "frame": 0,
+                            "points": [1.0, 2.0, 3.0, 2.0],
+                            "keyframe": True,
+                        },
+                        {
+                            "type": "rectangle",
+                            "frame": 3,
+                            "points": [1.0, 2.0, 3.0, 2.0],
+                            "outside": True,
+                        },
+                    ],
+                    "elements": [],
+                }
+            ],
+        }
+
+        original_annotations, imported_annotations = self.delete_annotation_and_import_annotations(
+            task_id, annotations, format_name, dataset_file
+        )
+
+        self.compare_original_and_import_annotations(original_annotations, imported_annotations)
+
+    @pytest.mark.parametrize("format_name", ["Datumaro 1.0", "COCO 1.0", "PASCAL VOC 1.1"])
+    def test_export_and_import_tracked_format_with_no_keyframe(self, format_name):
+        task_id = 14
+        dataset_file = self.tmp_dir / (format_name + "no_keyframe_source_data.zip")
+        annotations = {
+            "shapes": [],
+            "tracks": [
+                {
+                    "frame": 0,
+                    "shapes": [
+                        {
+                            "type": "rectangle",
+                            "frame": 0,
+                            "points": [1.0, 2.0, 3.0, 2.0],
+                        },
+                    ],
+                    "elements": [],
+                }
+            ],
+        }
+
+        original_annotations, imported_annotations = self.delete_annotation_and_import_annotations(
+            task_id, annotations, format_name, dataset_file
+        )
+
+        self.compare_original_and_import_annotations(original_annotations, imported_annotations)
```

Comment:
As above, it makes sense to add an extra check that there is the keyframe field set in the imported shape.